vid2cleantxt/transcribe.py

```diff
@@ -198,104 +227,232 @@ def save_transc_results(
+        ttext=full_text,
+        mdata=md_df,
+        verbose=verbose,
+    )
+
+    shutil.rmtree(ac_storedir, ignore_errors=True)
+    transc_res = {
+        "audio_transcription": full_transc,
+        "metadata": md_df,
+    }
+
+    if verbose:
+        print(f"finished transcription of {clip_name} - {get_timestamp()}")
+    logging.info(f"finished transcription of {clip_name} - {get_timestamp()}")
+    return transc_res
+
+
+def transcribe_video_wav2vec(
+    model,
+    processor,
+    clip_directory,
+    clip_name: str,
+    chunk_dur: int = 15,
+    temp_dir: str = "audio_chunks",
+    manually_clear_cuda_cache=False,
+    verbose=False,
+) -> dict:
+    """
+    transcribe_video_wav2vec - transcribe a video file using the wav2vec model
+
+    :param model: the model object
+    :param processor: the processor object
+    :param clip_directory: the directory of the video file
+    :param str clip_name: the name of the video file
+    :param int chunk_dur: the duration of each chunk in seconds, default 15
+    :param str temp_dir: the directory to store the audio chunks in. default "audio_chunks"
+    :param bool manually_clear_cuda_cache: whether to manually clear the cuda cache after each chunk. default False
+    :param bool verbose: whether to print the transcribed text locations to the console. default False
+    :return dict: a dictionary containing the transcribed text, the metadata
     """
     logging.info(f"Starting to transcribe {clip_name} @ {get_timestamp()}")
     if verbose:
         print(f"Starting to transcribe {clip_name} @ {get_timestamp()}")
-    # create audio chunk folder
-    ac_storedir = join(src_dir, temp_dir)
+    ac_storedir = join(clip_directory, temp_dir)
     create_folder(ac_storedir)
     use_attn = wav2vec2_islarge(
-        ts_model
+        model
     )  # if they pass in a large model, use attention masking
-    # get the audio chunks
+
     chunk_directory = prep_transc_pydub(
-        clip_name, src_dir, ac_storedir, chunk_dur, verbose=verbose
-    )
+        clip_name, clip_directory, ac_storedir, chunk_dur, verbose=verbose
+    )  # split the video into chunks
     torch_validate_cuda()
-    gc.collect()  # free up memory
-    device = "cuda:0" if torch.cuda.is_available() else "cpu"  # set device
+    gc.collect()
+    device = "cuda" if torch.cuda.is_available() else "cpu"  # set device
     full_transc = []
     GPU_update_incr = (
         math.ceil(len(chunk_directory) / 2) if len(chunk_directory) > 1 else 1
     )
+    model = model.to(device)
     pbar = tqdm(total=len(chunk_directory), desc="Transcribing video")
     for i, audio_chunk in enumerate(chunk_directory):
 
         # note that large-960h-lv60 has an attention mask of length of the input sequence, the base model does not
         if (i % GPU_update_incr == 0) and (GPU_update_incr != 0):
-            # provide update on GPU usage
-            check_runhardware()
+            check_runhardware()  # check utilization
```

Comment:
why not just use `torch.cuda.is_available()`

Reply:
good question, original idea was to print the RAM and stuff when you are transcribing on CPU (as I originally did). agree it is too verbose